### DISCUSSION TITLE
Trying to capture the output of vitest as a background process

### DISCUSSION THREAD

**Discussion by fgarcia** [vitest-dev/vitest#5366] [Q&A]
I was playing with a wrapper for Vitest to get some side actions if an error is printed (stderr). Everything seems to work fine, except that test names disappear from screen. Everything else works fine, even the screen redraws, but the lines showing each test are gone. 

Can anyone point me what I am missing out in this code?

```js
  import { spawn } from 'child_process'
  let child = spawn('npx', ['vitest'], {
    shell: true,
    stdio: ['inherit', 'pipe', 'pipe'],
    env: { FORCE_COLOR: '1', ...process.env },
  })
  child.stdout.on('data', dat => {
    process.stdout.write(dat)
  })
  child.stderr.on('error', dat => {
    process.stderr.write(dat)
  })

```

**Comment by AriPerkkio** [comment]
Vitest's internal test cases are using a bit similar setup. You could use it as reference.

https://github.com/vitest-dev/vitest/blob/6d1b145118d42b14bc0739820c4ce9e17fbd45dd/test/test-utils/index.ts#L51-L92

Make sure to use the `stripAnsi` on the captured output.
